src/amp-story-player/amp-story-player-impl.js

```diff
@@ -716,12 +731,83 @@ export class AmpStoryPlayer {
     new AmpStoryPlayerViewportObserver(this.win_, this.element_, () =>
       this.visibleDeferred_.resolve()
     );
+    if (DESKTOP_PANEL_STORY_PLAYER_EXP_ON) {
+      new this.win_.ResizeObserver((e) => {
+        const {width, height} = e[0].contentRect;
+        this.onPlayerResize_(width, height);
+      }).observe(this.element_);
+    }
 
     this.render_();
 
     this.element_.isLaidOut_ = true;
   }
 
+  /**
+   * Builds desktop "previous" and "next" story UI.
+   * @private
+   */
+  initializeDesktopStoryControlUI_() {
+    const prevButton = this.doc_.createElement('button');
+    prevButton.classList.add('i-amphtml-story-player-desktop-panel-prev');
+    prevButton.addEventListener('click', () => this.previous_());
+    prevButton.setAttribute('aria-label', 'previous story');
+    this.rootEl_.appendChild(prevButton);
+
+    const nextButton = this.doc_.createElement('button');
+    nextButton.classList.add('i-amphtml-story-player-desktop-panel-next');
+    nextButton.addEventListener('click', () => this.next_());
+    nextButton.setAttribute('aria-label', 'next story');
+    this.rootEl_.appendChild(nextButton);
+
+    const checkButtonsDisabled = () => {
+      prevButton.toggleAttribute(
+        'disabled',
+        this.isIndexOutofBounds_(this.currentIdx_ - 1) &&
+          !this.isCircularWrappingEnabled_
+      );
+      nextButton.toggleAttribute(
+        'disabled',
+        this.isIndexOutofBounds_(this.currentIdx_ + 1) &&
+          !this.isCircularWrappingEnabled_
+      );
+    };
+
+    this.element_.addEventListener('navigation', () => checkButtonsDisabled());
```

Comment:
Wdyt of having this here vs in the `onNavigation` method?

Reply:
It's more readable in `onNavigation` I'll change it to that.